crates/mev-share-rpc-api/src/types.rs

```diff
@@ -418,28 +418,97 @@ pub struct CancelPrivateTransactionRequest {
     pub tx_hash: H256,
 }
 
+// TODO(@optimiz-r): Revisit after <https://github.com/flashbots/flashbots-docs/issues/424> is closed.
 /// Response for `flashbots_getBundleStatsV2` represents stats for a single bundle
 ///
 /// Note: this is V2: <https://docs.flashbots.net/flashbots-auction/searchers/advanced/rpc-endpoint#flashbots_getbundlestatsv2>
 ///
 /// Timestamp format: "2022-10-06T21:36:06.322Z"
+#[derive(Default, Debug, Clone, PartialEq)]
+pub enum BundleStats {
+    /// The relayer has not yet seen the bundle.
+    #[default]
+    Unknown,
+    /// The relayer has seen the bundle, but has not simulated it yet.
+    Seen(StatsSeen),
+    /// The relayer has seen the bundle and has simulated it.
+    Simulated(StatsSimulated),
+}
+
+impl Serialize for BundleStats {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        match self {
+            BundleStats::Unknown => serde_json::json!({"isSimulated": false}).serialize(serializer),
+            BundleStats::Seen(stats) => stats.serialize(serializer),
+            BundleStats::Simulated(stats) => stats.serialize(serializer),
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for BundleStats {
+    fn deserialize<D>(deserializer: D) -> Result<BundleStats, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let map = serde_json::Map::deserialize(deserializer)?;
+
+        if map.get("receivedAt").is_none() {
+            Ok(BundleStats::Unknown)
+        } else if map["isSimulated"] == "false" {
```

Comment:
this doesn't look like serialize and deserialize is symmetrical.
looking at `isSimulated`